airbyte-integrations/connectors/source-github/source_github/source.py

```diff
@@ -229,3 +236,48 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
             WorkflowJobs(parent=workflow_runs_stream, **repository_args_with_start_date),
             TeamMemberships(parent=team_members_stream, **repository_args),
         ]
+
+
+class GithubAvailabilityStrategy(HTTPAvailabilityStrategy):
+    def check_availability(self, stream: Stream) -> Tuple[bool, any]:
+        stream_slice = self._get_stream_slice(stream)
+        # get out the stream_slice parts for later use.
+        organisation = stream_slice.get("organization", "")
+        repository = stream_slice.get("repository", "")
+        try:
+            records = stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice)
+            next(records)
+            return True, None
```

Comment:
You'll notice this is highly inspired by the ConnectionChecker class. I have about 30% of a mind that this all belongs in another method within that class 🤔 interested in thoughts from others